components/expression/expression-renderer.js

```diff
@@ -244,180 +413,341 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
+        this.signal('pathHover', this.data.profiles[i]);
+      }.bind(this))
+      .on('mouseout', function(value) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathUnhover', this.data.profiles[i]);
+      }.bind(this))
+      .on('click', function(value, i) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathClick', this.data.profiles[i]);
+      }.bind(this));
+  }, this);
 };
 
 /**
- * Computes the horizontal and vertical label sizes for the heatmap.
- * The results are stored into:
- *     this.geneLabelWidth,
- *     this.conditionLabelHeight
+ * Adds the expression profiles for the selected genes as line charts.
  * @private
  */
-genotet.ExpressionRenderer.prototype.getHeatmapLabelSizes_ = function() {
-  // TODO(liana): Computes the horizontal/vertical text margin allocated for
-  // the heatmap labels.
-  if (!this.data.options.showGeneLabels) {
-    this.geneLabelWidth_ = 0;
-  } else {
-    // To implement... replace dummy value.
-    this.geneLabelWidth_ = 30;
-  }
-  if (!this.data.options.showConditionLabels) {
-    this.conditionLabelHeight_ = 0;
-  } else {
-    // To implement... replace dummy value.
-    this.conditionLabelHeight_ = 50;
-  }
+genotet.ExpressionRenderer.prototype.addGeneProfile_ = function(geneIndex) {
+  var path = this.path_ = {
+    row: geneIndex
+  };
+  this.data.profiles.push(path);
+  this.drawGeneProfiles_();
 };
 
-/** @inheritDoc */
-genotet.ExpressionRenderer.prototype.resize = function() {
-  this.base.resize.call(this);
-  this.render();
+/**
+ * Adds the expression profiles for the selected genes as line charts.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.removeGeneProfile_ = function(geneIndex) {
+  var index = -1;
+  this.data.profiles.forEach(function(path, i) {
+    index = (path.row == geneIndex ? i : -1);
+  }, this);
+  //var index = this.data.profiles.indexOf(geneIndex);
+  this.data.profiles.splice(index, 1);
+  this.drawGeneProfiles_();
 };
 
-/*
-LayoutHeatmap.prototype.prepareLine = function() {
-  this.lines = this.parentView.viewdata.lineData;
-  this.maxval = this.tfaMaxval = 0;
-  this.minval = this.tfaMinval = 1E10;
-  for (var i = 0; i < this.lines.length; i++) {
-    this.maxval = Math.max(Math.max.apply(null, this.lines[i].values), this.maxval);
-    this.minval = Math.min(Math.min.apply(null, this.lines[i].values), this.minval);
-    for (var j = 0; j < this.lines[i].tfaValues.length; j++) {
-      this.tfaMaxval = Math.max(this.tfaMaxval, this.lines[i].tfaValues[j].value);
-      this.tfaMinval = Math.min(this.tfaMinval, this.lines[i].tfaValues[j].value);
-    }
-    this.lines[i].color = this.lineColors[i];
+/**
+ * Highlights the hover cell for the heatmap.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.highlightHoverCell_ = function(cell, highlight_) {
+  var cellSelection = d3.select(cell.container_);
+  if (highlight_) {
+    cellSelection.style('stroke', 'white');
```

Comment:
Re-classing every cell's 'text-highlight' to highlight a single cell doesn't seem to be a good idea...